gltflib/gltf.py

```diff
@@ -334,6 +352,17 @@ def convert_to_external_resource(self, resource: GLTFResource, uri: str) -> Exte
             resource.uri = uri
             return resource
 
+    @classmethod
+    def _decode_str(cls: 'GLTF', data: bytes, encoding: str = None) -> str:
+        if encoding is not None:
+            return data.decode(encoding, errors='replace')
+        elif data.startswith(codecs.BOM_UTF16_BE):
+            return data.decode('utf-16-be', errors='replace').lstrip('\ufeff')
+        elif data.startswith(codecs.BOM_UTF16_LE):
+            return data.decode('utf-16-le', errors='replace').lstrip('\ufeff')
+        else:
+            return data.decode('utf-8-sig', errors='replace')
```

Comment:
This would by default go to `utf-8-sig`, while it should've go to `utf-8` unless specified otherwise?

Reply:
Or maybe even
```
        if encoding is not None:
            return data.decode(encoding, errors='replace')
        elif data.startswith(codecs.BOM_UTF16_BE):
            return data.decode('utf-16-be', errors='replace').lstrip('\ufeff')
        elif data.startswith(codecs.BOM_UTF16_LE):
            return data.decode('utf-16-le', errors='replace').lstrip('\ufeff')
        elif data.startswith(codecs. BOM_UTF8):
            return data.decode('utf-8-sig', errors='replace')
        else:
            return data.decode('utf-8', errors='replace')
```
?

Reply:
`utf-8-sig` can be used to decode UTF-8 with or without BOM just fine, there's no need to split those as separate cases. If there is no BOM, it will still work and just read it as a normal UTF-8 file. If there is a BOM, it will get automatically stripped out when decoding (so no need for `lstrip` in that case, and it would have been a different byte sequence anyway).

It is perhaps more intention-revealing to have them as separate cases... but when researching this, I found it's a quite common recommendation to just read files using `utf-8-sig` instead of `utf-8`, as it can conveniently handle both. I'll add a comment to clarify the intent, so it doesn't look like an oversight.

Not sure why UTF-16 works differently, but the BOM does get preserved when decoding in that case, hence the special handling and manually stripping out the BOM sequence (and keeping the BOM in there causes `json.loads` to throw).

Reply:
Indeed, I've read up through the specs and that's how it was designed to work. Awesome!